core/controllers/feedback.py

```diff
@@ -241,10 +240,11 @@ def post(self):
             if len(message_text) > 200:
                 message_text = message_text[:200] + '...'
 
-            if exploration.title in messages:
-                messages[exploration.title].append(message_text)
+            reference_dict = {exploration.title: message_text}
+            if exploration.id in messages:
+                messages[exploration.id].update(reference_dict)
```

Comment:
I don't think this is correct. Wouldn't you just overwrite the existing message?

I suggest you format the 'messages' dict as:

    exploration.id: {
        'title': ...,
        'messages': [],  # append to this list
    }
```